test/pytest/mocktransport.py

```diff
@@ -54,15 +54,17 @@ async def check_open(
         channel=None,
         problem=None,
         reply_keys: Optional[JsonObject] = None,
+        absent_keys: Optional[List] = None,
```

Comment:
Cool addition!

Please use a fully-specified type, though: `List` on its own isn't enough.

Also: my personal preference these days is to write the type signatures in "modern" format, quoting if required for backwards compatibility.  eg:

```python
  absent_keys: 'list[str] | None' = None
```

That being said: I think list is the wrong type for this key anyway.  Something more like `Iterable[str]` with a default value of `()` seems nice?